src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -536,7 +536,7 @@ private string GetSiblingNamesToRemoveFromUnshippedTextCore(ISymbol symbol)
                         {
                             if (sibling.IsImplicitlyDeclared)
                             {
-                                if (!sibling.IsConstructor())
+                                if (!((sibling as IMethodSymbol)?.MethodKind is MethodKind.Constructor or MethodKind.PropertyGet or MethodKind.PropertySet))
```

Comment:
Something like this would be easier to read.

```suggestion
                                if (sibling is not IMethodSymbol { MethodKind: MethodKind.Constructor or MethodKind.PropertyGet or MethodKind.PropertySet })
```